api/src/main/java/org/pmiops/workbench/reporting/insertion/BigQueryInsertionPayloadTransformer.java

```diff
@@ -0,0 +1,14 @@
+package org.pmiops.workbench.reporting.insertion;
+
+/*
+ * Base interface for BigQuery payload builders. Parameterized class is a payload model type,
+ * and the intent is to pair each implementation with an enum type inheriting from
+ * QueryParameterColumn<T>.
+ *
+ * Builder interfaces for separate BigQuery upload paths are possible.
+ */
+public interface BigQueryInsertionPayloadTransformer<MODEL_TYPE> {
```

Comment:
opt: Having a long generic type name like this looks a bit unusual to me. We don't have an official style ruling on this, but FWIW Oracle's guides recommend a single character here.

Reply:
Regarding this one, yes single-character names have been standard forever. I thought it would make things easier to follow particularly in places where there are multiple type parameters). How about MODEL_T? Or is that worse?